### PR TITLE
Release version handling

### PR DESCRIPTION
**Pull request by str4d**
- Convert Zcash versions to Debian format for packaging
- [manpage] Handle build numbers in versions

### CODE REVIEW

**Comment by str4d** [comment]
This still needs testing for the tagged release case, where a Git version doesn't appear in the version output - but we never run `gen-manpages.sh` on a tagged commit, so this PR is probably sufficient for us.

**Comment by nathan-at-least** [comment]
If I have time (a big if) I'm going to work on some of the release automation tickets like this. If I need help or have to postpone this, so be it.

**Comment by nathan-at-least** [comment]
I had assigned myself, but it looks like @str4d has already done this. Reviewing now.

File: contrib/devtools/gen-manpages.sh [hunk 1]
```diff
@@ -11,7 +11,9 @@ ZCASHTX=${ZCASHTX:-$SRCDIR/zcash-tx}
 [ ! -x $ZCASHD ] && echo "$ZCASHD not found or not executable." && exit 1
 
 # The autodetected version git tag can screw up manpage output a little bit
-ZECVER=($($ZCASHCLI --version | head -n1 | awk -F'[ -]' '{ print $5, $6 }'))
+ZECVERSTR=$($ZCASHCLI --version | head -n1 | awk '{ print $NF }')
+ZECVER=$(echo $ZECVERSTR | awk -F- '{ OFS="-"; NF--; print $0; }')
+ZECCOMMIT=$(echo $ZECVERSTR | awk -F- '{ print $NF }')
```
Comment (nathan-at-least) [inline contrib/devtools/gen-manpages.sh]: Phew. Took a bit of reading awk and interactively running these with `v1.0.8-1` to know what this means. :-o

File: contrib/devtools/gen-manpages.sh [hunk 2]
```diff
@@ -21,8 +23,8 @@ $ZCASHD --version | sed -n '1!p' >> footer.h2m
 
 for cmd in $ZCASHD $ZCASHCLI $ZCASHTX; do
   cmdname="${cmd##*/}"
-  help2man -N --version-string=${ZECVER[0]} --include=footer.h2m -o ${MANDIR}/${cmdname}.1 ${cmd}
-  sed -i "s/\\\-${ZECVER[1]}//g" ${MANDIR}/${cmdname}.1
+  help2man -N --version-string=$ZECVER --include=footer.h2m -o ${MANDIR}/${cmdname}.1 ${cmd}
+  sed -i "s/\\\-$ZECCOMMIT//g" ${MANDIR}/${cmdname}.1
```
Comment (nathan-at-least) [inline contrib/devtools/gen-manpages.sh]: Does this mean for release `v1.0.8-1` the man page will refer to `v1.0.8` instead?

**Comment by bitcartel** [comment]
I thought all PRs had to have a corresponding Github issue ticket?

**Comment by bitcartel** [comment]
Partial ACK.

I took tip, went through dummy release process with version number 1.0.8-2.  Man pages were created okay.

I installed devscripts so that I could access dch on Ubuntu.

```
dch -v $DEBVERSION -D jessie -c contrib/debian/changelog
dch warning: Recognised distributions are: unstable, testing, stable,
oldstable, experimental, {testing-,stable-,oldstable-,}proposed-updates,
{testing,stable,oldstable}-security, wheezy-backports, jessie-backports and UNRELEASED.
Using your request anyway.
dch: Did you see that warning?  Press RETURN to continue...
```

After pressing RETURN, an editor is opened and enter a message for the change.  Note that instead of showing 1.0.8+2 it shows 1.0.8-2

```
diff --git a/contrib/debian/changelog b/contrib/debian/changelog
index cffa580..7ee84b6 100644
--- a/contrib/debian/changelog
+++ b/contrib/debian/changelog
@@ -1,3 +1,9 @@
+zcash (1.0.8-2) jessie; urgency=medium
+
+  * Hello 
+
+ -- Zcash Company <team@z.cash>  Thu, 18 May 2017 14:39:18 -0700
+
 zcash (1.0.8+1) jessie; urgency=high
 
   * 1.0.8-1 release.

```

**Comment by str4d** [comment]
@bitcartel that version is entirely set by `$DEBVERSION`. Did you use the new release process instructions to create it? ie:
```
export DEBVERSION=$(echo $ZCASH_RELEASE | sed 's/-beta/~beta/' | sed 's/-rc/~rc/' | sed 's/-/+/')
```

**Comment by bitcartel** [comment]
@str4d Ok, that works.  Btw, where are the new instructions?  I don't see it in this branch or on the current tip.

**Comment by nathan-at-least** [comment]
Hm, I tested with `./zcutil/build-debian-package.sh`. I think we should remove the redundant stuff from `release-process.md` and have it just say `Step XXX. run ./zcutil/build-debian-package.sh`. I'll file a ticket about cleaning up those instructions during this release.

File: zcutil/build-debian-package.sh [hunk 3]
```diff
@@ -18,6 +18,7 @@ if [ ! -d $BUILD_PATH ]; then
 fi
 
 PACKAGE_VERSION=$($SRC_PATH/src/zcashd --version | grep version | cut -d' ' -f4 | tr -d v)
+DEBVERSION=$(echo $PACKAGE_VERSION | sed 's/-beta/~beta/' | sed 's/-rc/~rc/' | sed 's/-/+/')
```
Comment (str4d) [inline zcutil/build-debian-package.sh]: @bitcartel do you see this line?

File: doc/release-process.md [hunk 4]
```diff
@@ -81,7 +81,7 @@ Add the newly created release notes to the Git repository:
 
 Update the Debian package changelog:
 
-    export DEBVERSION="${ZCASH_RELEASE}"
+    export DEBVERSION=$(echo $ZCASH_RELEASE | sed 's/-beta/~beta/' | sed 's/-rc/~rc/' | sed 's/-/+/')
```
Comment (str4d) [inline doc/release-process.md]: @bitcartel Do you see this line?

**Comment by nathan-at-least** [comment]
@zkbot r+

**Pull request merged**